common/testing/updateutils/update.go

```diff
@@ -0,0 +1,161 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package updateutils
+
+import (
+	"github.com/stretchr/testify/require"
+	commandpb "go.temporal.io/api/command/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	failurepb "go.temporal.io/api/failure/v1"
+	protocolpb "go.temporal.io/api/protocol/v1"
+	updatepb "go.temporal.io/api/update/v1"
+	"go.temporal.io/server/common/payloads"
+	"go.temporal.io/server/common/testing/protoutils"
+	"go.temporal.io/server/common/testing/testvars"
+)
+
+type (
+	UpdateUtils struct {
+		t require.TestingT
+	}
+
+	helper interface {
+		Helper()
+	}
+)
+
+func New(t require.TestingT) UpdateUtils {
+	return UpdateUtils{
+		t: t,
+	}
+}
+
+func (u UpdateUtils) UpdateAcceptCommands(tv *testvars.TestVars, messageID string) []*commandpb.Command {
+	if th, ok := u.t.(helper); ok {
+		th.Helper()
+	}
+
+	return []*commandpb.Command{{
+		CommandType: enumspb.COMMAND_TYPE_PROTOCOL_MESSAGE,
+		Attributes: &commandpb.Command_ProtocolMessageCommandAttributes{ProtocolMessageCommandAttributes: &commandpb.ProtocolMessageCommandAttributes{
+			MessageId: tv.MessageID("update-accepted", messageID),
+		}},
+	}}
+}
+
+func (u UpdateUtils) UpdateCompleteCommands(tv *testvars.TestVars, messageID string) []*commandpb.Command {
+	if th, ok := u.t.(helper); ok {
+		th.Helper()
+	}
+	return []*commandpb.Command{
+		{
+			CommandType: enumspb.COMMAND_TYPE_PROTOCOL_MESSAGE,
+			Attributes: &commandpb.Command_ProtocolMessageCommandAttributes{ProtocolMessageCommandAttributes: &commandpb.ProtocolMessageCommandAttributes{
+				MessageId: tv.MessageID("update-completed", messageID),
+			}},
+		},
+	}
+}
+
+func (u UpdateUtils) UpdateAcceptCompleteCommands(tv *testvars.TestVars, messageID string) []*commandpb.Command {
+	if th, ok := u.t.(helper); ok {
+		th.Helper()
+	}
+	return append(u.UpdateAcceptCommands(tv, messageID), u.UpdateCompleteCommands(tv, messageID)...)
+}
+
+func (u UpdateUtils) UpdateAcceptMessages(tv *testvars.TestVars, updRequestMsg *protocolpb.Message, messageID string) []*protocolpb.Message {
+	if th, ok := u.t.(helper); ok {
+		th.Helper()
+	}
+	updRequest := protoutils.UnmarshalAny[*updatepb.Request](u.t, updRequestMsg.GetBody())
+
+	return []*protocolpb.Message{
+		{
+			Id:                 tv.MessageID("update-accepted", messageID),
+			ProtocolInstanceId: updRequest.GetMeta().GetUpdateId(),
+			SequencingId:       nil,
+			Body: protoutils.MarshalAny(u.t, &updatepb.Acceptance{
+				AcceptedRequestMessageId:         updRequestMsg.GetId(),
+				AcceptedRequestSequencingEventId: updRequestMsg.GetEventId(),
+				AcceptedRequest:                  updRequest,
+			}),
+		},
+	}
+}
+
+func (u UpdateUtils) UpdateCompleteMessages(tv *testvars.TestVars, updRequestMsg *protocolpb.Message, messageID string) []*protocolpb.Message {
```

Comment:
should this be read "UpdateComplete Messages" or "Update CompleteMessages"? The message you created is a sccessful result. The name does not reflect it.

Reply:
It is "CompleteUpdate Messages". I swapped "Update" and "Complete" word, because I want all these helpers to be shown up close to each other in intellisense. Probably I shouldn't embed `UpdateUtils` to test suite but include it as a field.